Modules/selectmodule.c

```diff
@@ -554,7 +548,20 @@ poll_poll(pollObject *self, PyObject *args)
             return NULL;
         }
 
-        deadline = _PyTime_GetMonotonicClock() + timeout;
+        if (timeout >= 0) {
+            deadline = _PyTime_GetMonotonicClock() + timeout;
+        }
+    }
+
+    /* On some OSes, typically BSD-based ones, the timeout parameter of the
+       poll() syscall, when negative, must be exactly INFTIM, where defined,
+       or -1. See issue 31334. */
+    if (ms < 0) {
```

Comment:
Nitpick: Can we change this to `ms < -1`? Since we set `ms = -1` in line 528, we shouldn't get any "invalid argument" exception if we don't pass a `timeout` value to `p.poll()`.

Reply:
@berkerpeksag what if a OS defines `INFTIM` different from `-1`? This ensures that we set `ms` to `INFTIM` in that case too. I can change it to `ms <= -1` if you think it's more readable.

On a side note, the documentation says only that the timeout parameter is in milliseconds, but doesn't say it must be an integer. I've noticed that passing a float works too and in the special case of passing a float in the range `(-1, 0)`, that is converted to `0` by `ms = _PyTime_AsMilliseconds(timeout, _PyTime_ROUND_CEILING);`. I've tested that this behaviour does not change with my patch, but is arguably the desired one.

Reply:
I concur with @volans-. It is better to keep `ms < 0`.

As for rounding problem, I just have opened bpo-31786. I didn't mentioned it here because seems this change doesn't change this behavior.

Reply:
@volans- fair enough, thanks for the explanation. I think `ms < 0` can stay as is, no need to change it to `ms <= -1`.